src/Compatibility/Microsoft.DotNet.ApiCompatibility/ValidationSuppressions/Suppression.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System;
+using System.Globalization;
+
+namespace Microsoft.DotNet.ValidationSuppression
+{
+    /// <summary>
+    /// Represents a Suppression for a validation error.
+    /// </summary>
+    public class Suppression : IEquatable<Suppression>
+    {
+        /// <summary>
+        /// The DiagnosticId representing the error to be suppressed.
+        /// </summary>
+        public string? DiagnosticId { get; set; }
+
+        /// <summary>
+        /// The target of where to suppress the <see cref="DiagnosticId"/>
+        /// </summary>
+        public string? Target { get; set; }
+
+        /// <summary>
+        /// Left operand of an APICompat comparison.
+        /// </summary>
+        public string? Left { get; set; }
+
+        /// <summary>
+        /// Right operand of an APICompat comparison.
+        /// </summary>
+        public string? Right { get; set; }
+
+        /// <inheritdoc/>
+        public bool Equals(Suppression? other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            StringComparer stringComparer = StringComparer.Create(CultureInfo.InvariantCulture, ignoreCase: true);
+
+            if (AreDifferent(DiagnosticId, other.DiagnosticId) ||
+                AreDifferent(Target, other.Target) ||
+                AreDifferent(Left, other.Left) ||
+                AreDifferent(Right, other.Right))
+            {
+                return false;
+            }
+
+            return true;
+
+            bool AreDifferent(string? first, string? second) 
+                => !(string.IsNullOrEmpty(first?.Trim()) && string.IsNullOrEmpty(second?.Trim()) || stringComparer.Equals(first?.Trim(), second?.Trim()));
```

Comment:
You don't need to check for null or empty right? string.Equals should just be able to handle that for equality?

Reply:
It is able to handle null, but what I want in there is for null == string.Empty which the comparer won't mark as equal. The reason I want null to be equal to string.Empty, is because I want the following two suppressions to be considered equal:
```xml
  <Suppression>
    <DiagnosticId>CP0001</DiagnosticId>
    <Target>T:A.B</Target>
    <Left></Left>
    <Right></Right>
  </Suppression>
  <Suppression>
    <DiagnosticId>CP0001</DiagnosticId>
    <Target>T:A.B</Target>
  </Suppression>
```

Reply:
This case is covered by this unit test case: https://github.com/dotnet/sdk/pull/18185/files#diff-3e46c16064bf399d853838ec1bedde39d07377d9a5fa4a2f61671d49f47beccdR14

Reply:
A ok, makes sense.